modules/core/04-channel/types/codec.go

```diff
@@ -50,3 +52,35 @@ func RegisterInterfaces(registry codectypes.InterfaceRegistry) {
 // The actual codec used for serialization should be provided to x/ibc/core/04-channel and
 // defined at the application level.
 var SubModuleCdc = codec.NewProtoCodec(codectypes.NewInterfaceRegistry())
+
+// UnpackAcknowledgement unpacks an Any into an Acknowledgement. It returns an error if the
+// Any can't be unpacked into a Acknowledgement.
+func UnpackAcknowledgement(any *codectypes.Any) (exported.Acknowledgement, error) {
+	if any == nil {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrUnpackAny, "protobuf Any message cannot be nil")
+	}
+
+	ack, ok := any.GetCachedValue().(exported.Acknowledgement)
+	if !ok {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrUnpackAny, "cannot unpack Any into Acknowledgement %T", any)
+	}
+
+	return ack, nil
+}
+
+// PackAcknowledgement constructs a new Any packed with the given acknowledgement value. It returns
+// an error if the acknowledgement can't be casted to a protobuf message or if the concrete
+// implemention is not registered to the protobuf codec.
+func PackAcknowledgement(acknowledgement exported.Acknowledgement) (*codectypes.Any, error) {
+	msg, ok := acknowledgement.(proto.Message)
+	if !ok {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrPackAny, "cannot proto marshal %T", acknowledgement)
+	}
+
+	anyAcknowledgement, err := codectypes.NewAnyWithValue(msg)
+	if err != nil {
+		return nil, sdkerrors.Wrap(sdkerrors.ErrPackAny, err.Error())
```

Comment:
I know this was copied/pasted, but I think the better way of returning the error is:
```suggestion
		return nil, sdkerrors.Wrap(err, "failed to pack acknowledgement into Any")
```
If you agree maybe we could open up a good first issue to update all our other pack functions?